### PR TITLE
Enhance test coverage and configure Codecov JUnit reporting

### PR DESCRIPTION
**Pull request by jeremiah-k**
## Overview

This pull request expands automated test coverage across the codebase, improves test reliability and concurrency handling in tests, and updates CI to produce and upload JUnit XML test results to Codecov. It also includes small formatting and import adjustments in a few modules and test modernizations to make tests more robust.

## Key changes

Features
- CI: Pytest now emits JUnit XML (--junitxml=junit.xml) and a Codecov test-results upload step was added (pinned action commit). junit.xml was added to .gitignore and pytest.ini sets junit_family = legacy.
- Tests: New and expanded tests added across many modules to increase coverage and validate behavior under concurrency, edge cases, and deprecation paths.

Fixes (test reliability / concurrency)
- MeshInterface tests: added many thread-safety and concurrency tests (packet ID generation, node DB access, queue ops, response handler registration, safe close behavior, showNodes/getNode, sendText) and replaced fragile sleeps with thread events.
- MTConfig tests: added tests for reset behavior, clearing warned deprecations, compat-alias warn-once semantics, getattr/setattr/delattr coverage, default state validation, and concurrent access/thread-safety checks.
- TCPInterface tests: added reconnect/backoff tests, abort/early-exit scenarios, fatal-disconnect idempotence, and timing/cleanup behavior.
- SupportedDevice tests: extensive validation of USB ID parsing/normalization/aliases, equality semantics, device list contents, and edge cases.
- Util tests: coverage for DeferredExecution, DotDict/dotdict, our_exit, toNodeNum, toStr, flagsToList.

Refactors / test modernizations
- Replaced time.sleep usage with threading events where appropriate, removed daemon threads and added joins/locking to avoid races in tests, reorganized imports and added docstrings, replaced placeholder assertions with concrete checks, and improved deprecation-warn testing.
- Minor formatting changes: multi-line import formatting in test_node.py and wrapping a long assignment in meshtastic/node.py (no behavioral change).

Breaking changes / migration notes

- No production API changes or behavior-breaking changes were introduced. The PR is additive (tests, CI changes) and contains only small formatting refactors; no migration steps are required.

### CODE REVIEW

**Comment by jeremiah-k** [comment]
@coderabbitai review

**Comment by jeremiah-k** [comment]
@coderabbitai review

**Comment by jeremiah-k** [comment]
@coderabbitai full review

File: meshtastic/tests/test_mesh_interface.py [hunk 1]
```diff
@@ -964,3 +965,301 @@ def test_timeago_fuzz(seconds: int) -> None:
...
+                    with iface._node_db_lock:
+                        node = iface._get_or_create_by_num(node_num)
+                        node["lastHeard"] = i
+                        if iface.nodes is not None:
+                            iface.nodes[node_id] = node
+                        if iface.nodesByNum is not None:
+                            iface.nodesByNum[node_num] = node
+            except Exception as e:  # noqa: BLE001 - intentionally recording thread errors
+                with errors_lock:
+                    errors.append(e)
```
[the remark this answers is not in the document]
Comment (jeremiah-k) [reply meshtastic/tests/test_mesh_interface.py:1025]: @coderabbitai full review +

the direction to avoid nested lock acquisition is good, and we applied it in the test.

One correction: `_node_db_lock` is currently `threading.RLock` (re-entrant), so the specific “will deadlock with nested acquisition” claim is not accurate in current code. We still refactored to avoid nested locking for consistency with the lock-partition policy and clearer test intent. Examine the current code in full. Sync your understanding with the latest implementation, update/remove any stale learnings.
